components/server/src/util/request-context.ts

```diff
@@ -152,6 +170,14 @@ export function runWithRequestContext<T>(context: RequestContextSeed, fun: () =>
     return runWithContext({ ...context, requestId, startTime, cache }, fun);
 }
 
+export function toHeaders(headers: IncomingHttpHeaders): Headers {
```

Comment:
Could this go into `util/express`, maybe? :thinking: Reasoning: feels weird to have the dependecy on IncomingHttpHeaders here.